examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -14,36 +14,76 @@ import { customerServicePort } from "../mock-service-interface";
 import type { ITask, ITaskEvents, ITaskList, TaskData } from "../model-interface";
 
 class Task extends TypedEventEmitter<ITaskEvents> implements ITask {
-	public get id(): string {
-		return this._id;
-	}
-	// Probably would be nice to not hand out the SharedString, but the CollaborativeInput expects it.
-	public get name(): SharedString {
-		return this._name;
-	}
-	public get priority(): number {
-		const cellValue = this._priority.get();
-		if (cellValue === undefined) {
-			throw new Error("Expected a valid priority");
-		}
-		return cellValue;
-	}
-	public set priority(newValue: number) {
-		this._priority.set(newValue);
-	}
-	public constructor(
-		private readonly _id: string,
-		private readonly _name: SharedString,
-		private readonly _priority: ISharedCell<number>,
-	) {
-		super();
-		this._name.on("sequenceDelta", () => {
-			this.emit("nameChanged");
-		});
-		this._priority.on("valueChanged", () => {
-			this.emit("priorityChanged");
-		});
-	}
+    public get id(): string {
+        return this._id;
+    }
+    // Probably would be nice to not hand out the SharedString, but the CollaborativeInput expects it.
+    public get name(): SharedString {
+        return this._name;
+    }
+    public get priority(): number {
+        const cellValue = this._priority.get();
+        if (cellValue === undefined) {
+            throw new Error("Expected a valid priority");
+        }
+        return cellValue;
+    }
+    public get incomingName(): string | undefined {
+        return this._incomingName;
+    }
+    public set incomingName(newValue: string | undefined) {
+        this._incomingName = newValue;
+        this.emit('incomingNameChanged');
+    }
+    public get incomingType(): string | undefined {
+        return this._incomingType;
+    }
+    public set incomingType(newValue: string | undefined) {
+        this._incomingType = newValue;
+    }
+    public get incomingPriority(): number | undefined {
+        return this._incomingPriority;
+    }
+    public set incomingPriority(newValue: number | undefined) {
+        this._incomingPriority = newValue;
+        this.emit('incomingPriorityChanged');
+    }
+    public constructor(
+        private readonly _id: string,
+        private readonly _name: SharedString,
+        private readonly _priority: ISharedCell<number>,
+        private _incomingName: string | undefined,
+        private _incomingPriority: number | undefined,
+        private _incomingType: string | undefined,
```

Comment:
I might just not be getting how the model works, but having the incoming values as constructor parameters feels weird? When creating a task locally they wouldn't be required, and when dealing with incoming changes I'd expect that I'd be applying them to an existing task through its setters.